flag.go

```diff
@@ -40,6 +40,14 @@ type FlagLoader struct {
 
 	// Args defines a custom argument list. If nil, os.Args[1:] is used.
 	Args []string
+
+	// UsageFunc an optional function that is called to set a flag.Usage value
+	//  The input is the raw flag name, and the output should be a string
+	//  that will used in passed into the flag for Usage.
+	UsageFunc func(name string) string
+
+	// only exists for testing.  This is the raw flagset that is to parse
+	flagSet *flag.FlagSet
```

Comment:
If we go this way, we should stop passing the `flagSet` to other functions as well. So:

```go
func (f *FlagLoader) processField(flagSet *flag.FlagSet, fieldName string, field *structs.Field)
```

should be changed too

```go
func (f *FlagLoader) processField(fieldName string, field *structs.Field)
```

thoughts?